test/react-util/global-id.spec.ts

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { expect } from 'test-drive-react';
+import { getLocalId, getRootId, separator } from '../../src/react-util/global-id';
+
+interface TestProps {
+    id?: string;
+}
+
+class TestClass extends React.Component<TestProps> {}
+
+describe('GlobalID', () => {
+   describe('getRootId', () => {
+       const id = 'pancake';
+
+        it('resolves the id from props in case it was passed', () => {
+            const testClass = new TestClass({id});
+
+            expect(getRootId(testClass)).to.equal(id);
+        });
+
+        it('resolves the id from a component\'s private state (if relevant) if it was not provided in props', () => {
+            const firstInstance = new TestClass();
+            expect(getRootId(firstInstance)).to.equal('0');
+
+            const providedIdInstance = new TestClass({id});
+            expect(getRootId(providedIdInstance)).to.equal(id);
+
+            const incrementedIdInstance = new TestClass();
+            expect(getRootId(incrementedIdInstance)).to.equal('1');
+            expect(getRootId(firstInstance)).to.equal('0');
+        });
+
```

Comment:
other missing tests:

passing a props object to the function:
 -  result should be valid
 -  function should be pure (freeze the argument)